src/lib/support/SerializationUtils.cpp

```diff
@@ -998,12 +998,23 @@ WEAVE_ERROR ReadDataForType(TLVReader &aReader, void *aStructureData, const Fiel
 static WEAVE_ERROR FindNullifiedFieldsArray(void *aStructureData, const SchemaFieldDescriptor *aSchemaDescriptor, uint8_t *&aNullifiedFields)
 {
     uint32_t offset;
-    WEAVE_ERROR err;
+    WEAVE_ERROR err = WEAVE_NO_ERROR;
     const FieldDescriptor *lastFieldDescriptor = &(aSchemaDescriptor->mFields[aSchemaDescriptor->mNumFieldDescriptorElements - 1]);
+    bool lastFieldIsArray = false;
 
-    err = GetArrayElementSize(offset, lastFieldDescriptor, lastFieldDescriptor->GetType());
-    SuccessOrExit(err);
-
+    if (aSchemaDescriptor->mNumFieldDescriptorElements >= 2)
+    {
+        lastFieldIsArray = (aSchemaDescriptor->mFields[aSchemaDescriptor->mNumFieldDescriptorElements - 2].GetType() == SerializedFieldTypeArray);
+    }
+    if (lastFieldIsArray)
+    {
+        offset = sizeof(void *);
```

Comment:
Per chat with Lukasz: absent some archaic decorations on pointers (`far`, `near`, `huge`), all data pointers in C are the same size.  C99 makes allowances for the function pointers being of different size than `void *` but that distinction has no impact here.